rust/operator-binary/src/controller.rs

```diff
@@ -845,16 +849,123 @@ fn build_node_rolegroup_statefulset(
+///
+fn build_reporting_task_job(
+    nifi: &NifiCluster,
+    rolegroup_ref: &RoleGroupRef<NifiCluster>,
+) -> Result<Job> {
+    let rolegroup_obj_name = rolegroup_ref.object_name();
+    let namespace: &str = &nifi.namespace().context(ObjectHasNoNamespaceSnafu)?;
+    let nifi_connect_url = format!(
+        "https://{rolegroup}-0.{rolegroup}.{namespace}.svc.cluster.local:{port}/nifi-api",
+        rolegroup = rolegroup_obj_name,
+        namespace = namespace,
+        port = HTTPS_PORT
+    );
+
+    let mut container = ContainerBuilder::new("create-reporting-task")
+        .image(STACKABLE_TOOLS_IMAGE)
+        .command(vec!["sh".to_string(), "-c".to_string()])
+        .args(vec![[
+            "python/create_nifi_reporting_task.py",
+            &format!("-n {}", nifi_connect_url),
+            &format!("-u $(cat {}/username)", AUTH_VOLUME_MOUNT_PATH),
+            &format!("-p $(cat {}/password)", AUTH_VOLUME_MOUNT_PATH),
+            &format!("-v {}", nifi_version(nifi)?),
+            &format!("-m {}", METRICS_PORT),
+            &format!("-c {}/ca.crt", KEYSTORE_REPORTING_TASK_MOUNT),
+        ]
+        .join(" ")])
+        // The VolumeMount for the secret operator key store certificates
+        .add_volume_mount(KEYSTORE_VOLUME_NAME, KEYSTORE_REPORTING_TASK_MOUNT)
+        .security_context(SecurityContext {
+            run_as_user: Some(0),
+            ..SecurityContext::default()
+        })
+        .build();
+
+    // The Volume for the secret operator key store certificates
+    let mut volumes = vec![build_keystore_volume(KEYSTORE_VOLUME_NAME)];
+
+    // Volume and Volume mounts for the authentication secret
+    let auth_volumes = get_auth_volumes(&nifi.spec.authentication_config.method)
+        .context(MaterializeAuthConfigSnafu)?;
+
+    for (name, (mount_path, volume)) in auth_volumes {
+        container
+            .volume_mounts
+            .get_or_insert_with(Vec::default)
+            .push(VolumeMount {
+                mount_path: mount_path.to_string(),
+                name: name.to_string(),
+                ..VolumeMount::default()
+            });
+        volumes.push(volume);
+    }
+
+    let job_name = format!(
+        "{}-create-reporting-task-{}",
+        nifi.name(),
+        nifi_version(nifi)?.replace('.', "-")
+    );
+
+    let pod = PodTemplateSpec {
+        metadata: Some(
+            ObjectMetaBuilder::new()
+                .name(job_name.clone())
+                .namespace_opt(nifi.namespace())
+                .build(),
+        ),
+        spec: Some(PodSpec {
+            containers: vec![container],
+            // We use "OnFailure" here instead of "Never" to avoid spawning pods and pods. We just
+            // restart the existing pod in case the script fails
+            // (e.g. because the NiFi cluster is not ready yet).
+            // TODO: Do we want to add a final number of fails or a timeout?
```

Comment:
regarding the TODO: I think a final number of fails/timeout might be good? If it's not too difficult

Reply:
yeah its a little further down, the todo can go
```
    let job = Job {
        metadata: ObjectMetaBuilder::new()
            .name(job_name)
            .namespace_opt(nifi.namespace())
            .ownerreference_from_resource(nifi, None, Some(true))
            .context(ObjectMissingMetadataForOwnerRefSnafu)?
            .build(),
        spec: Some(JobSpec {
            backoff_limit: Some(100),
            ttl_seconds_after_finished: Some(120),
            template: pod,
            ..Default::default()
        }),
        status: None,
```

Reply:
:+1: